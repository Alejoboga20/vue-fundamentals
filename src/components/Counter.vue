<template lang="">
	<div class="counter">
		<h2>Counter</h2>
		<p>{{ counter }}<sup>2</sup> = {{ counter * counter }}</p>
	</div>
</template>

<script lang="ts">
export default {
	name: 'Counter',
	data() {
		return {
			counter: 10,
		};
	},
};
</script>

<style scoped>
.counter {
	text-align: center;
}
</style>
