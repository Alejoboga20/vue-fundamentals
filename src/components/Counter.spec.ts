import { shallowMount } from '@vue/test-utils';
import Counter from './Counter.vue';

describe('Counter.vue', () => {
	test('should match the snapshot', () => {
		const wrapper = shallowMount(Counter);

		expect(wrapper.html()).toMatchSnapshot();
	});
});
