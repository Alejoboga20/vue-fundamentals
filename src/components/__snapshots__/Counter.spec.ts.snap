// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter.vue > should match the snapshot 1`] = `
"<div data-v-9bd91872=\\"\\" class=\\"counter\\">
  <h2 data-v-9bd91872=\\"\\">Counter</h2>
  <p data-v-9bd91872=\\"\\">0<sup data-v-9bd91872=\\"\\">2</sup> = 0</p>
  <div data-v-9bd91872=\\"\\" class=\\"counter__buttons\\"><button data-v-9bd91872=\\"\\">+1</button><button data-v-9bd91872=\\"\\">-1</button></div>
</div>"
`;
